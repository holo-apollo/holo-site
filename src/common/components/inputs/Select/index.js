// @flow
import React from 'react';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import SelectMaterial from '@material-ui/core/Select';

import type { SelectOption } from 'common/types';
import FieldWithError from '../FieldWithError';

type Props = {
  options: SelectOption<any>[],
  label: string,
  errorText?: string,
};

const Select = ({ label, options, errorText, ...rest }: Props) => (
  <FieldWithError errorText={errorText}>
    <FormControl fullWidth={true} error={Boolean(errorText)}>
      <InputLabel error={Boolean(errorText)} style={{ fontSize: 12 }}>
        {label}
      </InputLabel>
      <SelectMaterial native={true} defaultValue={''} {...rest}>
        <option value={''} disabled={true} />
        {options.map(option => (
          <option key={option.value} value={option.value}>
            {option.label}
          </option>
        ))}
      </SelectMaterial>
    </FormControl>
  </FieldWithError>
);

export default Select;
