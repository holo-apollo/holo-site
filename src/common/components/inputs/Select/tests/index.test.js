import React from 'react';
import renderer from 'react-test-renderer';

import Select from '../index';

const options = [
  { value: 1, label: 'One' },
  { value: 2, label: 'Two' },
  { value: 3, label: 'Three' },
];

describe('Select', function() {
  it('should match snapshot', function() {
    const tree = renderer
      .create(<Select options={options} label="Label" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
