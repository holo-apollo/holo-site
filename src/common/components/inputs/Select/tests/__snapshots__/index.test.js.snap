// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match snapshot 1`] = `
<div
  className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
>
  <label
    className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
    data-shrink={false}
    style={
      Object {
        "fontSize": 12,
      }
    }
  >
    Label
  </label>
  <div
    className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31"
    onClick={[Function]}
  >
    <div
      className="MuiSelect-root-23"
    >
      <select
        aria-invalid={false}
        className="MuiSelect-select-24 MuiInputBase-input-53 MuiInput-input-38"
        defaultValue=""
        disabled={false}
        multiple={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
      >
        <option
          disabled={true}
          value=""
        />
        <option
          value={1}
        >
          One
        </option>
        <option
          value={2}
        >
          Two
        </option>
        <option
          value={3}
        >
          Three
        </option>
      </select>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-60 MuiSelect-icon-29"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
